contracts/vesting-lp/src/msg.rs

```diff
@@ -10,3 +17,129 @@ pub struct InstantiateMsg {
     /// Token info manager address
     pub token_info_manager: String,
 }
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Claim claims vested tokens and sends them to a recipient
+    Claim {
+        /// The address that receives the vested tokens
+        recipient: Option<String>,
+        /// The amount of tokens to claim
+        amount: Option<Uint128>,
+    },
+    /// Receives a message of type [`Cw20ReceiveMsg`] and processes it depending on the received template
+    Receive(Cw20ReceiveMsg),
+    /// RegisterVestingAccounts registers vesting targets/accounts
+    RegisterVestingAccounts {
+        vesting_accounts: Vec<VestingAccount>,
+    },
+    /// Creates a request to change contract ownership
+    /// ## Executor
+    /// Only the current owner can execute this
+    ProposeNewOwner {
+        /// The newly proposed owner
+        owner: String,
+        /// The validity period of the offer to change the owner
+        expires_in: u64,
+    },
+    /// Removes a request to change contract ownership
+    /// ## Executor
+    /// Only the current owner can execute this
+    DropOwnershipProposal {},
+    /// Claims contract ownership
+    /// ## Executor
+    /// Only the newly proposed owner can execute this
+    ClaimOwnership {},
+    /// Sets vesting token
+    /// ## Executor
+    /// Only the current owner or token info manager can execute this
+    SetVestingToken { vesting_token: AssetInfo },
+    /// Contains messages associated with the managed extension for vesting contracts.
+    ManagedExtension { msg: ExecuteMsgManaged },
+    /// Contains messages associated with the with_managers extension for vesting contracts.
+    WithManagersExtension { msg: ExecuteMsgWithManagers },
+    /// Contains messages associated with the historical extension for vesting contracts.
+    HistoricalExtension { msg: ExecuteMsgHistorical },
+    #[serde(rename = "migrate_liquidity_to_pcl_pool")]
+    MigrateLiquidityToPCLPool { user: Option<String> },
+    /// Callbacks; only callable by the contract itself.
+    Callback(CallbackMsg),
+}
+
+#[cw_serde]
+pub enum CallbackMsg {
+    MigrateLiquidityToClPair {
+        xyk_pair: Addr,
+        xyk_lp_token: Addr,
+        amount: Uint128,
+        slippage_tolerance: Decimal,
+        cl_pair: Addr,
+        ntrn_denom: String,
+        paired_asset_denom: String,
+        user: VestingAccountResponse,
+    },
+    ProvideLiquidityToClPairAfterWithdrawal {
+        ntrn_denom: String,
+        ntrn_init_balance: Uint128,
+        paired_asset_denom: String,
+        paired_asset_init_balance: Uint128,
+        cl_pair: Addr,
+        slippage_tolerance: Decimal,
+        user: VestingAccountResponse,
+    },
+    PostMigrationVestingReschedule {
+        user: VestingAccountResponse,
+        init_balance_pcl_lp: Uint128,
+    },
```

Comment:
why use 2 fields if you can actually use 1? i like it better

Reply:
if the point is to shorten the amount of fields I suggest to consider creating a dedicated structure containing all the fields for a single migration callback or reusing structures which meaning is appropriate. for example, you could reuse the Asset struct and put denom+amount inside a single struct:
```rust
pub enum CallbackMsg {
    ProvideLiquidityToClPairAfterWithdrawal {
        ntrn_init_balance: Asset,
        paired_asset_init_balance: Asset,
        cl_pair: Addr,
        slippage_tolerance: Decimal,
        user: VestingAccountResponse,
    }
}

pub struct Asset {
    /// Information about an asset stored in a [`AssetInfo`] struct
    pub info: AssetInfo,
    /// A token amount
    pub amount: Uint128,
}

pub enum AssetInfo {
    /// Non-native Token
    Token { contract_addr: Addr },
    /// Native token
    NativeToken { denom: String },
}
```

but VestingAccountResponse is designed for another use: represent the response model for QueryMsg::VestingAccount. I understand that currently the response model contains all the fields that you need in this case and seduction to reuse could be sensible, but it hypothetically could be so that the response model will be mutated (shortened in fields or expanded with other fields), and you won't be able to use it. this is of course not the case here but that's obviously a bad practice — to use things in the way they are not designed to be used.